sqlite_data_norm.c

```diff
@@ -0,0 +1,303 @@
+			return 0;
+		}
+	}
+	if (s[0] == '}')
+		s++;
+	return s[0] == 0;
+}
+
+/*
+ * uuid_generate generates a version 4 UUID as a string
+ *
+ *static void uuid_generate(sqlite3_context* context, int argc, sqlite3_value** argv)
+ *{
+ *   unsigned char aBlob[16];
+ *   unsigned char zs[37];
+ *   sqlite3_randomness(16, aBlob);
+ *   aBlob[6] = (aBlob[6] & 0x0f) + 0x40;
+ *   aBlob[8] = (aBlob[8] & 0x3f) + 0x80;
+ *   sqlite_fdw_data_norm_uuid_blob_to_str(aBlob, zs);
+ *   sqlite3_result_text(context, (char*)zs, 36, SQLITE_TRANSIENT);
+ *}
+ */
+
+/*
+ * uuid_str converts a UUID X into a well-formed UUID string.
+ * X can be either a string or a blob.
+ *
+ * static void uuid_str(sqlite3_context* context, int argc, sqlite3_value** argv) {
+ *	unsigned char aBlob[16];
+ *	unsigned char zs[37];
+ *	const unsigned char* pBlob;
+ *	(void)argc;
+ *	pBlob = sqlite_fdw_data_norm_uuid_input_to_blob(argv[0], aBlob);
+ *	if (pBlob == 0)
+ *		return;
+ *	sqlite_fdw_data_norm_uuid_blob_to_str(pBlob, zs);
+ *	sqlite3_result_text(context, (char*)zs, 36, SQLITE_TRANSIENT);
+ *}
+ */
+
+/*
+ * uuid_blob normalize text or blob UUID argv[0] into a 16-byte blob.
+ */
+static void
+sqlite_fdw_data_norm_uuid(sqlite3_context* context, int argc, sqlite3_value** argv)
+{
+	unsigned char aBlob[16];
+	sqlite3_value* arg = argv[0];		
+	
+	if (sqlite3_value_type(argv[0]) == SQLITE3_TEXT)
+	{
+		const unsigned char* txt = sqlite3_value_text(arg);
+		if (sqlite_fdw_uuid_blob(txt, aBlob))
+		{
+			sqlite3_result_blob(context, aBlob, 16, SQLITE_TRANSIENT);
+			return;
+		}
+	}
+	sqlite3_result_value(context, arg);
+}
+
+/*
+ * ISO:SQL valid boolean values with text affinity such as Y, no, f, t, oN etc.
+ * will be treated as boolean like in PostgreSQL console input
+ */ 
+static void
+sqlite_fdw_data_norm_bool(sqlite3_context* context, int argc, sqlite3_value** argv)
+{
+
+	sqlite3_value* arg = argv[0];
+	int dt = sqlite3_value_type(arg);
+	const char* t;
+	int l;
+
+	if (dt == SQLITE_INTEGER)
+	{
+		/* The fastest call because expected very often */
+		sqlite3_result_value(context, arg);
+		return;
+	}
+	if (dt != SQLITE3_TEXT && dt != SQLITE_BLOB )
+	{
+		/* NULL, FLOAT */
+		sqlite3_result_value(context, arg);
+		return;
+	}
+	l = sqlite3_value_bytes(arg);
+	if (l > 5)
+	{
+		sqlite3_result_value(context, arg);
+		return;
+	}
+	
+	t = (const char*)sqlite3_value_text(arg);
+		
+	if ( l == 1 )
+	{
+		if (strcasecmp(t, "t") == 0)
+		{
+			sqlite3_result_int(context, 1);
+			return;
+		}
+		if (strcasecmp(t, "f") == 0)
+		{
+			sqlite3_result_int(context, 0);
+			return;
+		}
+		if (strcasecmp(t, "y") == 0)
+		{
+			sqlite3_result_int(context, 1);
+			return;
+		}
+		if (strcasecmp(t, "n") == 0)
+		{
+			sqlite3_result_int(context, 0);
+			return;
+		}
+		/* rare but possible cases */
+		if (strcasecmp(t, "1") == 0)
+		{
+			sqlite3_result_int(context, 1);
+			return;
+		}
+		if (strcasecmp(t, "0") == 0)
+		{
+			sqlite3_result_int(context, 0);
+			return;
+		}
+	}
+	if ( l == 2 )
```

Comment:
I think it id better to use `else if`. Line 261, 274 and 279 are also same.

Reply:
Up to 3-4% productivity on 3Gb test. Thanks, @t-kataym! Fixed in https://github.com/pgspider/sqlite_fdw/pull/88/commits/72e1b22177db37251b2cda4474eb46955d1d6f26.

Reply:
Thank you for fixing.